### DISCUSSION TITLE
Can I set the replicacount on a PVC

### DISCUSSION THREAD

**Discussion by profhase** [longhorn/longhorn#7798] [General]
## Question

I would like to be able to set the number of replicas per PVC (for PVs created by longhorn).

Is there a way to do that via annotation?
Or do I have to specify a storage class for each number of replicas?

Thanks,
Ilya

## Environment

 - Longhorn version: newest
 - Kubernetes version: 1.25
 - Node config
   - OS type and version
   - Kernel version
   - CPU per node:
   - Memory per node:
   - Disk type
   - Network bandwidth and latency between the nodes:
 - Underlying Infrastructure (e.g. on AWS/GCE, EKS/GKE, VMWare/KVM, Baremetal):

## Additional context

Usually I use a Replicacount of 3 for security reasons. We do have some volumes for postgres backups (dumps) that are only there for snapshotting.

**Comment by c3y1huang** [comment]
Longhorn doesn't handle the replica count in the PVC.

If the volume has already been created, you can adjust the number of replicas directly in the Longhorn Volume. If you prefer the volume to be dynamically created with a specific replica count, you can define a new storage class with the desired replica count.

**Reply by profhase** [reply]
Thank you for the answer

Wouldn't it be convenient to allow to set a replicacount for the volume dynamically via annotation on PVC?

For now the solution is for us to have multiple storage classes:

`longhorn-rc1`, `longhorn-rc2`, `longhorn-rc3`

(for each replica count)
